docs/src/test/scala/sample/scaladsl/ConsumerExample.scala

```diff
@@ -341,3 +341,29 @@ object ExternallyControlledKafkaConsumer extends ConsumerExample {
   }
 }
 
+//Restart Kafka consumer
+object RestartKafkaConsumeExample extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+    // #restartConsumer
+    val runnableGraph: RunnableGraph[Consumer.Control] =
+      Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
+        .map(msg => { println(msg.record.value()); msg.committableOffset })
+        .map(_.commitScaladsl())
+        .toMat(Sink.ignore)(Keep.left)
+    val control1: Consumer.Control = runnableGraph.run()
+    val shutdownFuture = control1.shutdown()
+
+    shutdownFuture onComplete {
+      case Success(Done) =>
+        println("Stream was shutdown")
+
+        // Start the stream again
+        val control2: Consumer.Control = runnableGraph.run()
```

Comment:
@nafshartous Sounds good. It would also be good to include `future.onFailure` with handling that restarts the whole actor or the whole processing somehow (resend `Start` to itself?).
This is a very frequent question - how to deal with failing stream so souch example would be great.